src/data/equiv/fin.lean

```diff
@@ -44,6 +44,25 @@ def fin_succ_equiv (n : ℕ) : fin n.succ ≃ option (fin n) :=
     0 fin.succ : fin n.succ) = _, by rw fin.cases_succ) x,
   by rintro ⟨none | x⟩; [refl, exact fin.cases_succ _]⟩
 
+@[simp] lemma fin_succ_equiv_zero {n : ℕ} :
+  (fin_succ_equiv n) 0 = none := rfl
+
+@[simp] lemma fin_succ_equiv_succ {n : ℕ} (m : fin n):
+  (fin_succ_equiv n) m.succ = some m := by simp [fin_succ_equiv]
+
+@[simp] lemma fin_succ_equiv_ne_zero {n : ℕ} {m : fin (n + 1)} (h : m ≠ 0) :
+  (fin_succ_equiv n) m = some (m.pred h) :=
+begin
+  set m' : fin n := m.pred h,
+  rw [show m = (m.pred h).succ, by rw [fin.succ_pred], fin_succ_equiv_succ]
+end
```

Comment:
This one seems like an odd simp lemma. Did we actually need it to simplify the determinant expression?

Reply:
Without it, the simp expression in the proof of `equiv.perm.decompose_fin_symm_of_refl` won't fire. If you prefer, this can be moved into the body of that proof.

Reply:
This can now be removed I think, although I don't want to touch this file for fear of recompilation until we're done trying to reduce 3x3 matrices